File: q4.c
//LARGEST PALINDROME PRODUCT
#include<stdio.h>
int ispalindrome(int);
void main()
{
     int flag=1,i,j;
     for( i=999;i>=100;i--)
     {
          for( j=999;j>=100;j--)
          {
               int k=i*j;
               int ans=ispalindrome(k);
               if(ans==0)
               {flag=0;break;}
          }
          if(flag==0)
               break;
     }
     printf(" %d * %d = %d ,%d is the largest palindrome number.",i,j,i*j,i*j);

}
int ispalindrome(int a)
{
     int k=a,s=0;
     while(a!=0)
     {
          int r=a%10;
          s=s*10+r;
          a/=10;
     }
     if(k==s)
          return 0;
     else
          return 1;

}

File: q4.java
package hello;

public class q4 {

	public static void main(String[] args)
	{
	int max=-1,i,j;
	     for( i=999;i>=100;i--)
	     {
	          for( j=100;j<=999;j++)
	          {
	               int k=i*j;
	               String ans=String.valueOf(k);
	               StringBuffer rev=new StringBuffer(ans);
	               rev.reverse();
	               if(ans.contentEquals(rev)&&k>max)
	               {max=k;}
	          }
	               
	     }
	     System.out.println(max+" is the largest palindrome number.");


	}


}
